Prometheus.HttpExporter.AspNetCore/HttpRequestDuration/HttpRequestDurationMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace Prometheus.HttpExporter.AspNetCore.HttpRequestDuration
+{
+    public class HttpRequestDurationMiddleware : HttpRequestMiddlewareBase<Histogram>
+    {
+        public HttpRequestDurationMiddleware(RequestDelegate next, Histogram histogram)
+         : base(histogram)
+        {
+            this.next = next ?? throw new ArgumentNullException(nameof(next));
+            this.requestDuration = histogram;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var stopWatch = new Stopwatch();
+            
+            stopWatch.Start();
+            await this.next(context);
```

Comment:
An exception from `next()` would make the duration not be measured here. I think we should measure duration even in case of an exception here.